### DISCUSSION TITLE
Custom tooltip label with stacked chart

### DISCUSSION THREAD

**Discussion by mkfreeman** [observablehq/plot#2189] [Q&A] [answered]
Moved from https://talk.observablehq.com/t/custom-tooltip-label-with-stacked-chart/9874. 

In [this notebook](https://observablehq.com/d/569820eaae4023f5), I'm using an `offset: normalize` stack transform to create a stacked bar chart. I want to have a tooltip that displays the specific label "Percent: ##", but without changing the underlying data, I can't figure out how to customize the label. 

```js
// Current approach
channels: {
        customY: {
          label: "Percent",
          value: (d) => {
            // The value of d has the data values before normalization
            return d.population;
          }
        }
      },
      tip: {
        // Display custom values, hide the auto generated values
        format: {
          customY: true,
          test: (d) => d.population, // doesn't do anything
          y: true,
          x: true,
          fill: true
        }
      }
```

Seems similar to [this topic](https://talk.observablehq.com/t/pointer-transforms-with-px-py-on-stacked-bar-charts/8302). 

Any thoughts?

EDITS:

I was able to get the label by changing the data (but that seems somewhat undesireable as a practice):

```js
 Plot.tip(
     // Change the data
      tidy.map((d) => {
        d.percent = d.population;
        return d;
      }),
      {
        ...Plot.pointer(
          Plot.stackY({
            x: "state",
            y: "percent",
            fill: "age",
            sort: { color: null, x: "-y" },
            offset: "normalize"
          })
        )
      }
    )
```

@Fil Had suggested an offset rather than a stack normalize, though that didn't update the label associated with the percentage:

```js
Plot.barY(
  tidy,
  Plot.normalizeY("sum", {
    x: "state",
    y: "population",
    z: "state", // the series for normalizeY
    fill: "age",
    sort: { color: null, x: "-y" },
    offset: "normalize", // stack normalization (based on *x*)
    tip: {
      channels: {
        population: { value: "population", label: "abs population" }
      }
    }
  })
).plot({ y: { percent: true } })
```

**Comment by Fil** [comment]
"offset" is part of the stacking procedure. My remark is that we can't read the normalized values from it (because they are y1 and y2, and what we want is their differences. Instead, we explicitly use a normalizeY transform, with a *z* to define the series to be normalized.

But now I see that this cancels the sorting by y `sort: { color: null, x: "-y" }`, which we want to apply on the absolute populations.

Can you indicate what you're expecting to see in terms of labels?

**Comment by mkfreeman** [comment]
Thanks @Fil - sorry for any confusion. My objective is to customize the label associated with the stacked and normalized y value. In the screenshot below, I'd like to change the label for "population".
<img width="483" alt="Screenshot 2024-10-07 at 11 35 24 AM" src="https://github.com/user-attachments/assets/0dfda4b4-b324-4352-b854-e20f92f2815b">

I thought that by doing the following I could change that label:

```js
channels: {
        customY: {
          label: "Percent",
          value: "population"
        }
      },
      tip: {
        // Display custom values, hide the auto generated values
        format: {
          customY: true
        }
      }
```

However, the value associated with my custom label `Percent` is _not_ the normalized percentage. You can see the example in [this cell](https://observablehq.com/d/569820eaae4023f5#cell-8)

**Reply by Fil** [reply, marked as answer]
In my example code it works:

```
Plot.barY(
  tidy,
  Plot.normalizeY("sum", {
    x: "state",
    y: "population",
    z: "state", // the series for normalizeY
    fill: "age",
    sort: { color: null, x: "-y" },
    offset: "normalize", // stack normalization (based on *x*)
    tip: {
      channels: {
        population: { value: "population", label: "Absolute population" } // 👈
      }
    }
  })
).plot({ y: { label: "Relative population (%)", percent: true } })  // 👈
```

**Reply by mkfreeman** [reply]
I think I was spinning too long on this one, I can see that by assigning the `{y: label:....}` that you can customize the label for the axis (simpler than creating a new channel!). And helpful to see how to do both 🙏
